src/polar/Pie.tsx

```diff
@@ -155,6 +157,26 @@ type PieComposedData = PieCoordinate & {
   sectors: PieSectorDataItem[];
   data: RealPieData[];
 };
+type PiePayloadInputProps = {
+  sectors: PieSectorDataItem[];
+  legendType?: LegendType;
+};
+
+const computeLegendPayloadFromPieData = ({ sectors, legendType }: PiePayloadInputProps): Array<LegendPayload> => {
+  return sectors.map(
+    (entry: PieSectorDataItem): LegendPayload => ({
+      type: legendType,
+      value: entry.name,
+      color: entry.fill,
+      payload: entry,
+    }),
+  );
+};
+
+function SetPiePayloadLegend(props: PiePayloadInputProps): null {
+  useLegendPayloadDispatch(computeLegendPayloadFromPieData, props);
```

Comment:
we can call this from the actual component later when we refactor Pie to a function yeah?

Reply:
Exactly. I am not yet ready to do that because I am afraid something somewhere reads Pie's `defaultProps`.

Reply:
cool beans